engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -115,6 +115,9 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
       context stop self
       stay()
     case Event(JobStoreWriteFailure(_, t), _) =>
+      // This is moderately bad: If we can't write this result to the database, and then undergo a system restart, we'd need
+      // to re-run this job.
```

Comment:
I guess the alternative is to hold back the `response` until we get the JobStoreAck, and if we fail to write a success to the JobStore, we actually fail the job?